test/gtest/ucs/test_mpool.cc

```diff
@@ -444,3 +454,36 @@ UCS_TEST_SKIP_COND_F(test_mpool, alloc_4g, RUNNING_ON_VALGRIND) {
 
     ucs_mpool_cleanup(&mp, 1);
 }
+
+UCS_TEST_SKIP_COND_F(test_mpool, fifo_mode, !ENABLE_DEBUG_DATA)
+{
+    const size_t elem_size         = 16;
+    const unsigned elems_per_chunk = 32;
+    const unsigned max_elems       = 128;
+    ucs_mpool_t mp;
+    std::vector<void*> objs;
+
+    ucs_status_t status = setup_mpool(&mp, elem_size, elems_per_chunk,
+                                      max_elems);
+    ASSERT_UCS_OK(status);
+
+    int old                    = ucs_global_opts.mpool_fifo;
```

Comment:
pls use `modify_config()` instead

Reply:
added push/modify/pop config and tested with/without fix